src/style-property-map.js

```diff
@@ -86,7 +86,23 @@
     if (!propertyDictionary().isSupportedProperty(property)) {
       throw new TypeError('Cannot use has method for ' + property + ' because it is not a supported CSS property');
     }
-    return !!this._styleObject[property]
+    return !!this._styleObject[property];
+  };
+
+  StylePropertyMap.prototype.getProperties = function() {
+    var specifiedPropertyIndex = 0;
+    var isSpecifiedProperty = true;
+    var propertyArray = [];
+
+    while (isSpecifiedProperty) {
+      if (this._styleObject[specifiedPropertyIndex] == '') {
+        break;
+      }
+      propertyArray[specifiedPropertyIndex] =
+        this._styleObject[specifiedPropertyIndex];
+      specifiedPropertyIndex++;
+    }
+    return propertyArray.sort();
```

Comment:
Does the spec require this to be sorted? If not, let's not bother sorting the properties.